project_timesheet_time_control/models/account_analytic_line.py

```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antonio Espinosa
+# © 2016 Pedro M. Baeza
+# © 2016 Sergio Teruel
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, fields, api
+from datetime import datetime
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = 'account.analytic.line'
+
+    date_time = fields.Datetime(default=fields.Datetime.now, string='Date')
+    folded = fields.Boolean(related='task_id.stage_id.fold')
+
+    @api.onchange('account_id')
+    def onchange_account_id(self):
+        if self.task_id.project_id.analytic_account_id != self.account_id:
+            self.task_id = False
+        domain = {'task_id': []}
+        if self.account_id:
+            project = self.env['project.project'].search(
+                [('analytic_account_id', '=', self.account_id.id)], limit=1)
+            domain = {'task_id': [('project_id', '=', project.id),
+                                  ('stage_id.fold', '=', False)]}
+        return {'domain': domain}
+
+    @api.onchange('task_id')
+    def onchange_task_id(self):
+        if self.task_id:
+            self.account_id = self.task_id.project_id.analytic_account_id.id
+
+    def eval_date(self, vals):
+        if 'date_time' in vals and 'date' not in vals:
+            vals['date'] = fields.Date.from_string(vals['date_time'])
+        return vals
```

Comment:
Since this is not a model specific feature, I would rather have it as a function outside the class.
I suggest the name `dates_from_strings()`.